sdk/typespec/src/error/mod.rs

```diff
@@ -71,7 +71,7 @@ impl Display for ErrorKind {
 /// An error encountered when communicating with the service.
 #[derive(Debug)]
 pub struct Error {
-    context: Context,
+    context: ErrorContext,
```

Comment:
Part of me wonders if the Azure::Core::Context is the type in Rust that has a poor name choice. Afterall the field on Error is already context.

We should user study this naming choice before GA

Reply:
This isn't even exposed. `Context` in the public API is fine and matches other contexts in other languages' HTTP stacks.